Library/Homebrew/formula.rb

```diff
@@ -567,7 +567,7 @@ def synced_with_other_formulae?
     params(name: String, klass: T.class_of(Resource), block: T.nilable(T.proc.bind(Resource).void))
       .returns(T.nilable(Resource))
   }
-  def resource(name, klass = Resource, &block) = active_spec.resource(name, klass, &block)
+  def resource(name = T.unsafe(nil), klass = T.unsafe(nil), &block) = active_spec.resource(*name, *klass, &block)
```

Comment:
The `*name` doesn't seem correct as `resource(nil, Resource)` would get passed back as `resource(Resource)`.

Sure, technically not valid for the signature but formulae aren't typechecked and this would miss the `name.nil?` check we have.

Reply:
This is only a pass-through method, we still check `name.nil?` in `SoftwareSpec#resource`.

Reply:
Yeah but it's not entirely passthrough as `resource(nil, Resource)` passes through as `resource(Resource)` due to the splat.

Reply:
```suggestion
  def resource(name = T.unsafe(nil), klass = T.unsafe(nil), &block)
    if klass.nil?
      active_spec.resource(*name, &block)
    else
      active_spec.resource(name, klass)
    end
  end
```